src/journal/journald-server.c

```diff
@@ -818,6 +818,65 @@ static bool shall_try_append_again(JournalFile *f, int r) {
         }
 }
 
+static void flush_failed_entries_logs(const FailedEntryStats* failed_entry, int level) {
+        if (failed_entry->num_entries == 0)
+                return;
+
+        const char* dest = failed_entry->file ? strjoina(" to file ", failed_entry->file) : "";
+
+        if (failed_entry->num_entries > 1)
+                log_full_errno(
+                                level,
+                                failed_entry->error,
+                                "Failed to write %zu entries%s (%zu items, %zu bytes)%s: %m",
+                                failed_entry->num_entries,
+                                dest,
+                                failed_entry->num_items,
+                                failed_entry->total_size,
+                                failed_entry->message);
+        else
+                log_full_errno(
+                                level,
+                                failed_entry->error,
+                                "Failed to write entry%s (%zu items, %zu bytes)%s: %m",
+                                dest,
+                                failed_entry->num_items,
+                                failed_entry->total_size,
+                                failed_entry->message);
+}
+
+static void log_failed_entry(
+                FailedEntryStats *failed_entry,
+                int level,
+                int _error,
+                const char* _file,
+                const char* _message,
+                size_t n,
+                size_t size) {
+        if (_error != failed_entry->error) {
+                flush_failed_entries_logs(failed_entry, level);
+                failed_entry->num_entries = 0;
+                failed_entry->num_items = 0;
+                failed_entry->total_size = 0;
+                ratelimit_reset(&failed_entry->ratelimit);
+                failed_entry->error = _error;
+        }
+
+        failed_entry->num_entries += 1;
+        failed_entry->num_items += n;
+        failed_entry->total_size += size;
```

Comment:
I am not sure the finegrained counting is really something we want or need. I think a message saying "Dropped 542 similar messages"  or so after we are leaving the ratelimit window is sufficient, and not saying what was in them

Reply:
Fixed in commit https://github.com/systemd/systemd/commit/d77fdd883eeadce347ca517d21b8cf2b1a2a6f04